kolibri/core/auth/management/commands/sync.py

```diff
@@ -158,13 +159,15 @@ def handle_async(self, *args, **options):
 
         # pull from server and push our own data to server
         if not no_pull:
-            sync_client.initiate_pull(Filter(dataset_id))
+            with db_task_write_lock:
+                sync_client.initiate_pull(Filter(dataset_id))
         if not no_push:
             sync_client.initiate_push(Filter(dataset_id))
```

Comment:
I think we need to wrap this as well because we serialize as part of a pull or push.